### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by patrick-rodgers**
Large refactoring, looking for comments from team

### CODE REVIEW

**Comment by patrick-rodgers** [comment]
Should also add this is not fully tested yet, wanted folks to have a chance to review while I am out. Will test and update as needed before merge.

File: src/sharepoint/rest/lists.ts [hunk 1]
```diff
@@ -4,16 +4,15 @@ import { Items } from "./items";
 import { Views, View } from "./views";
 import { ContentTypes } from "./contenttypes";
 import { Fields } from "./fields";
-import { Queryable } from "./queryable";
+import { Queryable, QueryableCollection, QueryableSecurable } from "./queryable";
 import * as Util from "../../utils/util";
-import * as Mixins from "./mixins";
-import { Gettable, FilterableSelectableGettable } from "./actionables";
+import { ITypedHash } from "../../collections/collections";
```
Comment (pbjorklund) [inline src/sharepoint/rest/lists.ts]: Refer to https://github.com/OfficeDev/PnP-JS-Core/issues/18 about naming convention for interfaces

Comment (patrick-rodgers) [inline src/sharepoint/rest/lists.ts]: Ya, agree - will update. Also good to start a code standards section of wiki with that link as a first thing. Thanks!

**Comment by alex-randall** [comment]
Regarding the **dist** folder removal, makes sense as we want to encourage folks to download the library through NPM if at all possible.  And... people can accidently commit the contents in pull requests and have out of sync versions of stuff.

However:

- some folks don't have internet access (a lot of my customers in public sector and systems integrators)
- some folks don't use editors to code JavaScript in SharePoint, some do it directly from the browser inside a Script Editor Web Part, or Content Editor Web Part.  These people could also benefit greatly from the library.
- some folks are not allowed to install software (including NPM) on their machines due to enterprise policy (a lot of my customers in public sector and systems integrators)

Perhaps instead:

Maybe dist folder is more for "local builds" (and can safely be excluded from the git commits), but we need a specialized folder for the "real output" for people to download the pnp.d.ts file and the .js/.map files for those scenarios above.  And, in future documentation, we could then link to those downloads directly but only for scenarios above (the majority of the documentation will always steer them the NPM way of course).

What does everyone think?

**Comment by patrick-rodgers** [comment]
Closing this to create a new PR with the ITypedHash change included.

**Pull request closed without merging**